regression-tests/pure2-bugfix-for-nested-lists.cpp2

```diff
@@ -0,0 +1,27 @@
+point: @value type = {
+    public x: int = 0;
+    public y: int = 0;
+    operator=: (implicit out this, x_: int, y_: int) = {
+        x = x_;
+        y = y_;
+    }
+}
+
+check: (p: point) p;
+
+main: () = {
+    assert(check((17, 29)).x == 17);
+    assert(check((17, 29)).y == 29);
+
+    board: std::array<std::array<u8, 3>, 3> = ((
+                                            ('O', 'X', 'O'),
+                                            (' ', ('X'), 'X'),
+                                            ('X', 'O', 'O')
+    ));
+    assert(board[0] == :std::array<u8, 3> = ('O', 'X', 'O'));
+    assert(board[1] == :std::array<u8, 3> = (' ', 'X', 'X'));
+    assert(board[2] == :std::array<u8, 3> = ('X', 'O', 'O'));
+
+    // Still parentheses
+    assert((:std::vector = (17, 29)).size() == 2);
```

Comment:
For some reason, a semicolon doesn't work to delimit the unnamed declaration:
```suggestion
    assert(:std::vector = (17, 29);.size() == 2);
```
Also, I think `(0).f()` should lower to `f({0})`, just like `f((0))` (despite <https://github.com/hsutter/cppfront/issues/542#issuecomment-1662319573>).

Reply:
Commit ab379fc which lowers interpolations wrapped in `std::string()` rather than just `()` to avoid an initializer list argument.
We may want to group the object argument expression in UFCS, but avoid an initializer list argument.
In this context, we could use `copy` (<https://github.com/hsutter/cppfront/issues/466#copy>) (or worse, `std::identity()()`).